cloudinit/config/schema.py

```diff
@@ -160,6 +161,8 @@ class SchemaType(Enum):
 
     CLOUD_CONFIG = "cloud-config"
     NETWORK_CONFIG = "network-config"
+    NETWORK_CONFIG_V1 = "network-config-v1"
```

Comment:
NETWORK_CONFIG is a generic enum that will eventually be transformed into NETWORK_CONFIG_V1 or NETWORK_CONFIG_V2.  I chose to retain it for consistent user experience since the network config version isn't automatically determinable until the config file is parsed.  Until the config file is parsed for the version, it's assumed to be V2.  I was pretty on the fence about this design choice -- we could also update the command line arguments to require network-config-v1 or network-config-v2 or alias network-config to one of those two.  But I thought that could be confusing as previously the use provided network-config only and the parsing of the schema automatically translated it into v1 vs v2.  Thoughts?